File: server/src/seed.ts
import mongoose from 'mongoose';
import dotenv from 'dotenv';
import User from './models/User';
import Job from './models/Job';

dotenv.config();

const seed = async () => {
  try {
    await mongoose.connect(process.env.MONGODB_URI as string);
    console.log('✅ Connected to MongoDB for seeding...');

    // Clear existing data
    await User.deleteMany({});
    await Job.deleteMany({});
    console.log('🗑️  Cleared existing data');

    // Create Hiring Manager
    const manager = await User.create({
      name: 'Ankit Kumar',
      email: 'manager@kalpana.com',
      password: 'Manager@123',
      role: 'hiring_manager',
      company: 'Kalpana Software Solution Pvt. Ltd.',
      location: 'Bengaluru, Karnataka',
      bio: 'Experienced hiring manager with 8+ years in tech recruitment.',
      phone: '+91 98765 43210',
    });

    // Create Student
    const student = await User.create({
      name: 'Rahul Sharma',
      email: 'student@kalpana.com',
      password: 'Student@123',
      role: 'student',
      location: 'Delhi, India',
      bio: 'Final year CSE student passionate about full-stack development.',
      phone: '+91 87654 32109',
    });

    // Create sample jobs
    const jobs = [
      {
        title: 'Full Stack Developer',
        company: 'Kalpana Software Solution Pvt. Ltd.',
        location: 'Bengaluru, Karnataka',
        description: `We are looking for a talented Full Stack Developer to join our growing engineering team.

Key Responsibilities:
- Build and maintain web applications using React and Node.js
- Design and implement RESTful APIs
- Collaborate with the design team to implement UI/UX
- Write clean, maintainable, and well-documented code
- Participate in code reviews and agile sprints

What We Offer:
- Competitive salary
- Health insurance
- Work from home 2 days/week
- Learning & development budget`,
        requirements: [
          'React.js / Next.js',
          'Node.js / Express.js',
          'MongoDB / PostgreSQL',
          'TypeScript',
          'Git & GitHub',
          '2+ years experience',
        ],
        salary: '₹8L - ₹15L per annum',
        type: 'full-time',
        postedBy: manager._id,
        source: 'internal',
        status: 'open',
      },
      {
        title: 'Frontend React Developer',
        company: 'Kalpana Software Solution Pvt. Ltd.',
        location: 'Remote',
        description: `Join our frontend team to build beautiful and performant user interfaces.

You will work on:
- Building reusable React components
- Implementing responsive designs
- Performance optimization
- Integration with backend APIs`,
        requirements: [
          'React.js',
          'TypeScript',
          'Tailwind CSS',
          'REST API integration',
          'Responsive design',
        ],
        salary: '₹6L - ₹10L per annum',
        type: 'remote',
        postedBy: manager._id,
        source: 'internal',
        status: 'open',
      },
      {
        title: 'Backend Node.js Developer',
        company: 'Kalpana Software Solution Pvt. Ltd.',
        location: 'Hyderabad, Telangana',
        description: `We need a backend developer to build robust APIs and microservices.

Responsibilities:
- Design and develop RESTful APIs
- Database design and optimization
- Authentication and security implementation
- API documentation`,
        requirements: [
          'Node.js',
          'Express.js',
          'MongoDB',
          'JWT Authentication',
          'Docker (nice to have)',
        ],
        salary: '₹7L - ₹12L per annum',
        type: 'full-time',
        postedBy: manager._id,
        source: 'internal',
        status: 'open',
      },
      {
        title: 'SDE-1 Internship (6 months)',
        company: 'Kalpana Software Solution Pvt. Ltd.',
        location: 'Bengaluru, Karnataka',
        description: `Exciting 6-month internship for final year students and fresh graduates.

What you'll do:
- Work on real production projects
- Learn from experienced engineers
- Get hands-on with modern tech stack
- Possibility of full-time conversion`,
        requirements: [
          'JavaScript / TypeScript',
          'React basics',
          'Any backend language',
          'Final year student or fresher',
        ],
        salary: '₹20,000 - ₹30,000/month',
        type: 'internship',
        postedBy: manager._id,
        source: 'internal',
        status: 'open',
      },
    ];

    await Job.insertMany(jobs);
    console.log('✅ Sample jobs created');

    console.log('\n🎉 Database seeded successfully!\n');
    console.log('Test Credentials:');
    console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━');
    console.log('👔 Hiring Manager:');
    console.log('   Email: manager@kalpana.com');
    console.log('   Password: Manager@123');
    console.log('');
    console.log('🎓 Student:');
    console.log('   Email: student@kalpana.com');
    console.log('   Password: Student@123');
    console.log('━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n');

    process.exit(0);
  } catch (error) {
    console.error('❌ Seed error:', error);
    process.exit(1);
  }
};

seed();

File: server/seed.py
"""
seed.py — Populate MongoDB with test data.

Usage:
    cd server
    python seed.py
"""

import asyncio
import os
from datetime import datetime
from dotenv import load_dotenv

load_dotenv()

from motor.motor_asyncio import AsyncIOMotorClient
from beanie import init_beanie

from app.models.user import User
from app.models.job import Job
from app.models.application import Application


async def seed():
    mongo_uri = os.getenv("MONGODB_URI", "mongodb://localhost:27017/job_scraper_platform")
    client = AsyncIOMotorClient(mongo_uri)
    db = client.get_default_database()
    await init_beanie(database=db, document_models=[User, Job, Application])
    print("[+] Connected to MongoDB for seeding...")

    # Clear existing data
    await User.delete_all()
    await Job.delete_all()
    await Application.delete_all()
    print("[+] Cleared existing data")

    # Create 1 Hiring Manager
    manager = User(
        name="Ankit Kumar",
        email="manager@jobsphere.com",
        password="ManagerPass#2026",
        role="hiring_manager",
        company="JobSphere Tech Pvt. Ltd.",
        location="Bengaluru, Karnataka",
        bio="Experienced hiring manager with 8+ years in tech recruitment.",
        phone="+91 98765 43210",
    )
    manager.hash_password()
    await manager.insert()

    # Create 1 Student
    student = User(
        name="Rahul Sharma",
        email="student@jobsphere.com",
        password="StudentPass#2026",
        role="student",
        location="Delhi, India",
        bio="Final year CSE student passionate about full-stack development.",
        phone="+91 87654 32109",
    )
    student.hash_password()
    await student.insert()

    # Create sample jobs
    jobs_data = [
        {
            "title": "Full Stack Developer",
            "company": "JobSphere Tech Pvt. Ltd.",
            "location": "Bengaluru, Karnataka",
            "description": (
                "We are looking for a talented Full Stack Developer to join our growing engineering team.\n\n"
                "Key Responsibilities:\n"
                "- Build and maintain web applications using React and Node.js\n"
                "- Design and implement RESTful APIs\n"
                "- Collaborate with the design team to implement UI/UX\n"
                "- Write clean, maintainable, and well-documented code\n"
                "- Participate in code reviews and agile sprints\n\n"
                "What We Offer:\n"
                "- Competitive salary\n"
                "- Health insurance\n"
                "- Work from home 2 days/week\n"
                "- Learning & development budget"
            ),
            "requirements": ["React.js / Next.js", "Node.js / Express.js", "MongoDB / PostgreSQL",
                             "JavaScript", "Git & GitHub", "2+ years experience"],
            "salary": "INR 8L - 15L per annum",
            "type": "full-time",
            "posted_by": str(manager.id),
            "source": "internal",
            "status": "open",
        },
        {
            "title": "Frontend React Developer",
            "company": "JobSphere Tech Pvt. Ltd.",
            "location": "Remote",
            "description": (
                "Join our frontend team to build beautiful and performant user interfaces.\n\n"
                "You will work on:\n"
                "- Building reusable React components\n"
                "- Implementing responsive designs\n"
                "- Performance optimization\n"
                "- Integration with backend APIs"
            ),
            "requirements": ["React.js", "JavaScript", "Tailwind CSS", "REST API integration", "Responsive design"],
            "salary": "INR 6L - 10L per annum",
            "type": "remote",
            "posted_by": str(manager.id),
            "source": "internal",
            "status": "open",
        },
        {
            "title": "Backend Python Developer",
            "company": "JobSphere Tech Pvt. Ltd.",
            "location": "Hyderabad, Telangana",
            "description": (
                "We need a backend developer to build robust APIs and microservices.\n\n"
                "Responsibilities:\n"
                "- Design and develop RESTful APIs with FastAPI\n"
                "- Database design and optimization\n"
                "- Authentication and security implementation\n"
                "- API documentation"
            ),
            "requirements": ["Python", "FastAPI", "MongoDB", "JWT Authentication", "Docker (nice to have)"],
            "salary": "INR 7L - 12L per annum",
            "type": "full-time",
            "posted_by": str(manager.id),
            "source": "internal",
            "status": "open",
        },
        {
            "title": "SDE-1 Internship (6 months)",
            "company": "JobSphere Tech Pvt. Ltd.",
            "location": "Bengaluru, Karnataka",
            "description": (
                "Exciting 6-month internship for final year students and fresh graduates.\n\n"
                "What you'll do:\n"
                "- Work on real production projects\n"
                "- Learn from experienced engineers\n"
                "- Get hands-on with modern tech stack\n"
                "- Possibility of full-time conversion"
            ),
            "requirements": ["JavaScript", "React basics", "Any backend language",
                             "Final year student or fresher"],
            "salary": "INR 20,000 - 30,000/month",
            "type": "internship",
            "posted_by": str(manager.id),
            "source": "internal",
            "status": "open",
        },
    ]

    for jd in jobs_data:
        job = Job(
            title=jd["title"],
            company=jd["company"],
            location=jd["location"],
            description=jd["description"],
            requirements=jd["requirements"],
            salary=jd["salary"],
            type=jd["type"],  # type: ignore[arg-type]
            posted_by=jd["posted_by"],
            source=jd["source"],
            status=jd["status"],  # type: ignore[arg-type]
        )
        await job.insert()

    print("[+] Sample jobs created")
    print("\n[+] Database seeded successfully!\n")
    print("Test Credentials:")
    print("=" * 45)
    print("Hiring Manager:")
    print("   Email: manager@jobsphere.com")
    print("   Password: ManagerPass#2026")
    print("")
    print("Student:")
    print("   Email: student@jobsphere.com")
    print("   Password: StudentPass#2026")
    print("=" * 45 + "\n")


if __name__ == "__main__":
    asyncio.run(seed())
